### PR TITLE
Flagging UI Components

### PR DESCRIPTION
**Pull request by gbashour**
- Socket IO connection working, Client and Server can communication
- Client connection is established when a Student joins the Exam Session
- Basic events are emitted

### CODE REVIEW

File: client/src/components/FlagNotification.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React, { useState, useEffect } from 'react';
+import { io } from 'socket.io-client';
+import "../css/FlagNotification.css";
+import WarningIcon from '@mui/icons-material/Warning';
+
+const FlagNotification = () => {
+  const [flagAdded, setFlagAdded] = useState(false); // For Student
+
+  useEffect(() => {
+    const socket = io('http://localhost:4001'); // Change the URL to match your Socket.io server's URL
```
Comment (abrarbarbhuyia) [inline client/src/components/FlagNotification.js]: This might need to be changed to the process.env.REACT_APP_SERVER_URL url

Comment (abrarbarbhuyia) [reply client/src/components/FlagNotification.js]: So REACT_APP_SERVER_URL + `:4001`

File: server/routes/flag.js [hunk 2]
```diff
@@ -5,6 +5,21 @@ const Flag = require('../models/flagModel');
 const databaseMaster = require('../DatabaseAccess/databaseMaster');
 const { v4: uuidv4 } = require('uuid');
 
+const PORT = 4001;
+
+const io = require('socket.io')(PORT, {
+    cors: {
+        origin: ['http://localhost:3000'] // client URL
```
Comment (abrarbarbhuyia) [inline server/routes/flag.js]: Might need to change to ['http://localhost:3000', 'https://sentinel-frontend.vercel.app']

**Review by abrarbarbhuyia — CHANGES_REQUESTED** [review]
GOOD

File: client/src/components/FlagNotification.js (hunk 1 above)
Comment (abrarbarbhuyia) [reply client/src/components/FlagNotification.js]: 4001 in single quotes tho

**Review by abrarbarbhuyia — APPROVED** [review]
lgtm

**Comment by gbashour** [comment]
Updates:
- Admin Notification functionality has been implemented and tested in Exam Session
- Socket IO emits occur in a connected user's individual room associated with their ID
- Styling for Notifications
- Integration for Admin and Student interaction

**Pull request merged**